src/v/storage/probe.cc

```diff
@@ -248,6 +251,19 @@ void probe::setup_metrics(const model::ntp& ntp) {
             "Number of segments that have been compacted away "
             "during adjacent merge compaction."),
           labels),
+        sm::make_histogram(
+          "segment_size",
+          ssx::rate_limited_function<
+            ss::metrics::histogram(),
+            seastar::lowres_clock,
+            config::binding<std::optional<std::chrono::milliseconds>>>(
+            [this] {
+                return _segment_histogram_cb().seastar_histogram_logform();
+            },
+            config::shard_local_cfg()
+              .storage_segment_size_refresh_rate_ms.bind()),
+          sm::description("Local segment size histogram in bytes."),
+          labels),
```

Comment:
Can we do this not per ntp but per broker or at least per shard? A histogram per ntp could be quite expensive in terms of metrics cardinality.